alicloud/data_source_alicloud_yundun_bastionhost_instances.go

```diff
@@ -0,0 +1,206 @@
+)
+
+func dataSourceAlicloudBastionhostInstances() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAlicloudBastionhostInstancesRead,
+
+		Schema: map[string]*schema.Schema{
+			"description_regex": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validateNameRegex,
+			},
+			"output_file": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"ids": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				ForceNew: true,
+			},
+			"descriptions": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"instances": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"description": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"user_vswitch_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"private_domain": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"public_domain": {
+							Type:     schema.TypeString,
+							Computed: true,
+							Optional: true,
+						},
+						"instance_status": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"license_code": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"public_network_access": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"security_group_ids": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceAlicloudBastionhostInstancesRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+
+	request := yundun_bastionhost.CreateDescribeInstanceBastionhostRequest()
+	request.PageSize = requests.NewInteger(PageSizeLarge)
+	request.CurrentPage = requests.NewInteger(1)
+	var instances []yundun_bastionhost.Instance
+
+	// get name Regex
+	var nameRegex *regexp.Regexp
+	if v, ok := d.GetOk("description_regex"); ok {
+		if r, err := regexp.Compile(v.(string)); err == nil {
+			nameRegex = r
+		}
+	}
+
+	if v, ok := d.GetOk("ids"); ok {
+		ids, _ := v.([]interface{})
+		var ids_str []string
+		for _, v_instance_id := range ids {
+			ids_str = append(ids_str, v_instance_id.(string))
+		}
+		request.InstanceId = &ids_str
+	}
+	for {
+		raw, err := client.WithBastionhostClient(func(bastionhostClient *yundun_bastionhost.Client) (interface{}, error) {
+			return bastionhostClient.DescribeInstanceBastionhost(request)
+		})
+		if err != nil {
+			return WrapErrorf(err, DataDefaultErrorMsg, "alicloud_yundun_bastionhost", request.GetActionName(), AlibabaCloudSdkGoERROR)
```

Comment:
please use `alicloud_yundun_bastionhost_instances` instead of `alicloud_yundun_bastionhost`